src/main/java/sirius/biz/storage/layer3/DeleteFilesJob.java

```diff
@@ -45,7 +50,81 @@ protected DeleteFilesJob(ProcessContext process) {
 
     @Override
     public void execute() throws Exception {
+        VirtualFile sourcePath = process.require(SOURCE_PATH_PARAMETER);
+        if (!sourcePath.exists() || !sourcePath.isDirectory()) {
+            process.log(ProcessLog.error()
+                                  .withNLSKey("DeleteFilesJob.sourcePath.invalid")
+                                  .withContext(PATH_KEY, sourcePath.path()));
+            return;
+        }
+        handleDirectory(sourcePath);
+    }
+
+    private boolean handleDirectory(VirtualFile directory) {
+        AtomicBoolean childSkipped = new AtomicBoolean(false);
+
+        directory.allChildren().excludeFiles().subTreeOnly().maxDepth(1).iterate(subDirectory -> {
+            if (process.isActive()) {
+                childSkipped.set(!handleDirectory(subDirectory));
+            } else {
+                childSkipped.set(true);
```

Comment:
**Optional:** In my opinion this is a bit misleading. Should we rather name the boolean something like `shouldAbort`?

Reply:
I think abort feels *too hard*, like in job aborted or the like. The var means exactly that we skipped a child folder or file, either due to error or condition.